File: 3-Patron/php/Tesoro.php
<?php
class Tesoro{
	private $bonus;
	private $monedas;

	public function __construct($bonus, $monedas){
    $this->bonus = $bonus;
    $this->monedas = $monedas;
	}

	public function getBonus(){
		return $this->bonus;	
	}

	public function setBonus($bonus){
		$this->bonus = $bonus;
	}

	public function getMonedas(){
		return $this->monedas;	
	}

	public function setMonedas($monedas){
		$this->monedas = $monedas;
	}

	public function incBonus(){
		$this->bonus += 1.01;
	}
}
?>

File: 3-Patron/Java/Tesoro.java
package buildercomposite;

public class Tesoro {
    private float Bonus;
    private int Monedas;
    
    public Tesoro(float bonus, int monedas){
        this.Bonus = bonus;
        this.Monedas = monedas;
    }

 
    
    public float getBonus(){
        return this.Bonus;
    }
    public void setBonus(float bonus){
        this.Bonus = bonus;
    }
    public int getMonedas(){
        return this.Monedas;
    }
    public void setMonedas(int monedas){
        this.Monedas = monedas;
    }
    public void incBonus(){
        this.Bonus += 1.01; 
    }
}
